src/core/cgroup.c

```diff
@@ -1062,15 +1063,30 @@ static void cgroup_context_apply(
 
 CGroupMask cgroup_context_get_mask(CGroupContext *c) {
         CGroupMask mask = 0;
+        bool need_cpumask_for_cpu_accounting;
+        struct utsname u;
+
+        /* In the unified cgroup hierarachy CPU accounting information is
+         * read from cpu.stat file. On Linux 4.15 and newer cpu.stat exists
+         * without CPU controller enabled. On the other hand, on older kernels
+         * CPU controller has to be enabled for cpu.stat to appear. Let's check it.
+         * Related discussion found at PR #9665. */
+        assert_se(uname(&u) >= 0);
+        need_cpumask_for_cpu_accounting = str_verscmp(u.release, "4.15") < 0;
```

Comment:
hmm, doing this on each call might be needlessly expensive... could you turn the above two lines into a function of its own, that uses an internal `static` variable to cache the result? If you need inspiration how that function should look like, please look for the definition of urlify_enabled() in the sources (as one example), which does this correctly...